src/blog/2024/09/how-to-scrape-web-data-with-node-red.md

```diff
@@ -0,0 +1,155 @@
+
+By automating data collection, web scraping helps save time and reduce human error. It enables users to extract and analyze structured data from various sources, making it easier to compile and utilize information for research, business intelligence, or other purposes.
+
+## Why Use Web Scraping If APIs Are Available?
+
+APIs are a convenient way to access data from websites, but not all sites offer APIs, or their APIs may not provide the specific data you need.
+
+Web scraping can be helpful when APIs are unavailable or do not meet your requirements. It allows you to collect data directly from web pages, which can be beneficial for tasks like competitive analysis, market research, or tracking specific online content.
+
+## How Web Scraping Works?
+
+Web scraping involves systematically extracting data from websites using automated tools or scripts. The process begins with sending an HTTP request to a website's server, requesting access to a specific webpage. In response, the server provides the HTML content of the page. This HTML code contains the structured information displayed on the webpage, organized in a format that describes the layout and content.
+
+Once the HTML is received, the next step is parsing it. Parsing involves analyzing the HTML structure to identify and extract the data of interest. This may include navigating through nested elements, locating specific tags, and using selectors to target precise content such as text blocks, images, or links. The extracted data is then processed and stored in a format that suits the user's needs, whether a database, a CSV file, or another format suitable for analysis.
+
+## Web scrapping with Node-RED 
+
+In this section, we will guide you through the process of scraping data from publicly available websites using Node-RED and demonstrate how to extract data from a website specifically designed for scraping practice. For this example, we will scrape country data from the page at `https://www.scrapethissite.com/pages/simple/.`
+
+### Sending Requests to a Webpage
+
+To start scraping data, follow these steps to send an HTTP GET request to the webpage:
+
+1. Drag the **inject** node onto the canvas. This node allows you to manually trigger the HTTP request or set it to fire at specific intervals.
+2. Drag the **http request** node onto the canvas. Double-click it to configure and set the **Method** to `GET.` Enter the webpage URL you want to scrape (e.g., `https://www.scrapethissite.com/pages/simple/`).
+3. Drag the **debug** node onto the canvas.
+4. Connect the **inject** node's output to the input of the **http request** node and the **http request** node's output to the input of the **debug** node.
+5. Click **Deploy** to save and deploy your flow.
```

Comment:
Why can't Node-RED directly scrape the data?

Reply:
Because getting the webpage HTML is the first step of web scraping, it needs to be parsed and transformed. First, I have shown how to get the webpage HTML, then how to parse specific parts of that HTML, and finally how to transform and structure it so that it is ready for analysis or further processing.

Reply:
@sumitshinde-84 There's custom nodes to do this, no need to depend on an external site. And otherwise there's ways to scrape with xpaths in raw HTML?

Reply:
We are not using external sites to scrape data. `scrapethissite.com` ( developed to test the scrapping)  is a site that allows scraping data from their own site. We can't use other public websites, as most of them have terms and conditions that don't allow scraping